presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java

```diff
@@ -158,6 +160,9 @@ public void testComparison()
         assertOptimizedEquals("bound_long = unbound_long", "1234 = unbound_long");
 
         assertOptimizedEquals("10151082135029368 = 10151082135029369", "false");
+
+        assertOptimizedEquals("bound_varbinary = X'a b'", "true");
```

Comment:
Martin:

actually this the grammar from the ISO_9075_02_Foundation_2011_E.pdf that
Haozhun passed to me, spaces are allowed anywhere in the quote:

"

<binary string literal> ::=
X <quote> [ <space>... ] [ { <hexit> [ <space>... ] <hexit> [ <space>... ]
}... ] <quote>

      [ { <separator> <quote> [ <space>... ] [ { <hexit> [ <space>... ]
      <hexit> [ <space>... ] }... ] <quote> }... ]

"


On Sat, Dec 19, 2015 at 11:56 AM, Martin Traverso <notifications@github.com>
wrote:

> In
> presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java
> <https://github.com/facebook/presto/pull/3997#discussion_r48096166>:
>
> > @@ -158,6 +160,9 @@ public void testComparison()
> >          assertOptimizedEquals("bound_long = unbound_long", "1234 = unbound_long");
> >
> >          assertOptimizedEquals("10151082135029368 = 10151082135029369", "false");
> > +
> > +        assertOptimizedEquals("bound_varbinary = X'a b'", "true");
>
> These are invalid binary literals. According to the grammar rules in the
> spec, the string between the quotes needs to be made of pairs of
> hexadecimal characters optionally separated by one or more spaces.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/presto/pull/3997/files#r48096166>.
>